File: algorithm/stack/150.cpp
#include <vector>
#include <string>
#include <stack>
using namespace std;

class Solution
{
public:
    int evalRPN(vector<string> &tokens)
    {
        stack<int> myStack;
        int t1, t2;
        for (const auto &t : tokens)
        {
            if (t == "+")
            {
                t1 = myStack.top();
                myStack.pop();
                t2 = myStack.top();
                myStack.pop();
                myStack.push(t2 + t1);
            }
            else if (t == "-")
            {
                t1 = myStack.top();
                myStack.pop();
                t2 = myStack.top();
                myStack.pop();
                myStack.push(t2 - t1);
            }
            else if (t == "*")
            {
                t1 = myStack.top();
                myStack.pop();
                t2 = myStack.top();
                myStack.pop();
                myStack.push(t2 * t1);
            }
            else if (t == "/")
            {
                t1 = myStack.top();
                myStack.pop();
                t2 = myStack.top();
                myStack.pop();
                myStack.push(int(t2 / t1));
            }
            else
            {
                myStack.push(stoi(t));
            }
        }
        return myStack.top();
    }
};

File: algorithm/stack/150.py
from typing import List

class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        myStack = []
        for t in tokens:
            if t == '+':
                t1 = myStack.pop()
                t2 = myStack.pop()
                myStack.append(t2 + t1)
            elif t == '-':
                t1 = myStack.pop()
                t2 = myStack.pop()
                myStack.append(t2 - t1)
            elif t == '*':
                t1 = myStack.pop()
                t2 = myStack.pop()
                myStack.append(t2 * t1)
            elif t == '/':
                t1 = myStack.pop()
                t2 = myStack.pop()
                myStack.append(int(t2 / t1))
            else:
                myStack.append(int(t))
        return myStack[-1]
                


if __name__ == "__main__":
    tokens = ["10","6","9","3","+","-11","*","/","*","17","+","5","+"]
    print(Solution().evalRPN(tokens=tokens))